workspaces/src/network/result.rs

```diff
@@ -53,3 +53,32 @@ impl From<FinalExecutionOutcomeView> for CallExecutionDetails {
         }
     }
 }
+
+/// The result from a call into a View function. This contains the contents or
+/// the results from the view function call itself. The consumer of this object
+/// can choose how to deserialize its contents.
+pub struct ViewResultDetails {
+    /// Our result from our call into a view function.
+    pub result: Vec<u8>,
+    /// Logs generated from the view function.
+    pub logs: Vec<String>,
+}
+
+impl ViewResultDetails {
+    pub fn try_serde_deser<'a, T: serde::Deserialize<'a>>(&'a self) -> anyhow::Result<T> {
```

Comment:
```suggestion
    pub fn json<T: serde::DeserializeOwned>(&self) -> anyhow::Result<T> {
```

Reply:
Why use DeserializeOwned when there can be references to the buffer in the deserialization? Am I missing something where there are no cases of this with json? But yes, I think naming for json to be more explicit would be best

Reply:
Mostly "because reqwest does this", but I can see several benefits here:

* Simplicity for the user
* Pit of success -- non-owned deserialize is a very niche thing, and I don't think it can ever be useful here. But I can see users spending extra mental cycles to make borrowed deserialization work simply because that's available. 
* THere's also angle that DeserializeOwned is more flexible for us, and would allow, eg, to receive bytes into a non-contiguous buffer. But that's a bad point, as we expose `Vec<u8>` anyway, and we *should* expose it for simplicity.

Reply:
No strong opinion though: `Deserialize<'a>` would be ok. I do feel much stronger about method's name though.

Reply:
My guess for reqwest doing this is maybe because they are deserializing from a stream rather than a buffer? 

Yeah I think it's definitely niche, but as to your last point: I also just don't think it will ever make sense in this example for part of the response to be some stream that we keep open until used and disallow using more than once.

I guess the argument against this could be that a user might not know they can use `DeserializeOwned` in place of this if they are creating some convenience function with generics, but this also seems like a niche case.

Might be bikeshedding at this point, so I'll just make it clear I'm fine with both ways, just figured I'd think through

Reply:
No strong opinions here either, but let's just go with simplicity for now and have `DeserializeOwned`. I'm using it already in the ref-finance example/PR for `CallExecutionDetails` so this will make it consistent with that too